editor/code_editor.cpp

```diff
@@ -419,6 +419,32 @@ void FindReplaceBar::_update_matches_label() {
 	}
 }
 
+void FindReplaceBar::_temp_unfold_line(int p_line) {
+	// Unfold p_line, then walk back to find which was the first line that had to be unfolded so p_line can be shown.
+	// Mark that line as temporarily unfolded, so it's folded back on the next temporarily unfolding - unless the user for example edits the text.
+	if (temp_unfolded_line != -1) {
+		text_editor->fold_line(temp_unfolded_line);
+	}
+
+	if (!text_editor->is_line_folded(p_line)) {
+		temp_unfolded_line = p_line;
+		while (temp_unfolded_line >= 0) {
+			if (text_editor->is_line_folded(temp_unfolded_line)) {
+				break;
+			} else {
+				String line_text = text_editor->get_line(temp_unfolded_line);
+				if (line_text.is_empty() || !is_whitespace(line_text[0])) {
```

Comment:
This will fail in a number of cases, like:
```gdscript
func test():
    foo()
#   bar()
    baz() <-
```
Or
```gdscript
# This
# Is
# A <-
# Comment
```

With the arrow indicating where search is

Since this is just for performance I think we can leave it to a more dedicated fix that handles different cases more carefully, for example as suggested in the `CodeEdit`

Reply:
Oh, yeah that's true - but `in_string` and `in_comment` checks should really solve all cases.